### PR TITLE
feat(console): use figma-like controls for the map view

### PR DESCRIPTION
**Pull request by skyrpex**
On MacOS:

- Move two fingers together to translate the map
- Use pinch gesture to zoom in and out
- If the map becomes out of bounds, a message will appear along with a button to fit the map to the screen

### CODE REVIEW

**Review by eladb — APPROVED** [review]

File: apps/wing-console/console/ui/src/ui/zoom-pane.tsx [hunk 1]
```diff
@@ -33,181 +23,287 @@ export interface Viewport {
...
+  };
+};
 
-const selectionTransition = (selection: Selection, duration?: number) => {
-  const newSelection = (selection as any).transition();
-  if (duration) {
-    return newSelection.duration(duration) as Selection;
-  }
-  return newSelection as Selection;
+const boundingBoxOverlap = (a: BoundingBox, b: BoundingBox) => {
```
Comment (polamoros) [inline apps/wing-console/console/ui/src/ui/zoom-pane.tsx:49]: could you add a comment here?

**Comment by polamoros** [comment]
Zoom in/ zoomm out is not centering the map anymore.
https://github.com/winglang/wing/assets/5547636/cc27d7ca-7f03-42ce-a9e6-421008a33699

**Comment by polamoros** [comment]
When is the "out of bounds" message showing up?

https://github.com/winglang/wing/assets/5547636/8f3ec7a8-b5eb-43b2-a1db-49c4b8158904

**Review by polamoros — CHANGES_REQUESTED** [review]

**Review by polamoros — APPROVED** [review]

**Comment by Chriscbr** [comment]
🚀 🚀 🚀

**Pull request merged**